lib/apiservers/engine/backends/container.go

```diff
@@ -1243,35 +1256,39 @@ func containerInfoToDockerContainerInspect(id string, info *models.ContainerInfo
 			}
 		}
 		if info.ContainerConfig.LayerID != nil {
-			inpsectJSON.Image = *info.ContainerConfig.LayerID
+			inspectJSON.Image = *info.ContainerConfig.LayerID
 		}
 		if info.ContainerConfig.LogPath != nil {
-			inpsectJSON.LogPath = *info.ContainerConfig.LogPath
+			inspectJSON.LogPath = *info.ContainerConfig.LogPath
 		}
 		if info.ContainerConfig.RestartCount != nil {
-			inpsectJSON.RestartCount = int(*info.ContainerConfig.RestartCount)
+			inspectJSON.RestartCount = int(*info.ContainerConfig.RestartCount)
 		}
 		if info.ContainerConfig.ContainerID != nil {
-			inpsectJSON.ID = *info.ContainerConfig.ContainerID
+			inspectJSON.ID = *info.ContainerConfig.ContainerID
 		}
 		if info.ContainerConfig.Created != nil {
-			inpsectJSON.Created = time.Unix(*info.ContainerConfig.Created, 0).String()
+			inspectJSON.Created = time.Unix(*info.ContainerConfig.Created, 0).String()
 		}
 		if len(info.ContainerConfig.Names) > 0 {
-			inpsectJSON.Name = info.ContainerConfig.Names[0]
+			inspectJSON.Name = info.ContainerConfig.Names[0]
 		}
 	}
 
-	return inpsectJSON, nil
+	return inspectJSON, nil
 }
 
-// hostConfigFromContainerInfo() extracts docker compatible hostconfig data from the
-// Swagger-based ContainerInfo object.
-func hostConfigFromContainerInfo(id string, info *models.ContainerInfo) *container.HostConfig {
-	if info == nil {
+// hostConfigFromContainerInfo() gets the hostconfig that is passed to the backend during
+// docker create and updates any needed info
+func hostConfigFromContainerInfo(vc *viccontainer.VicContainer, info *models.ContainerInfo) *container.HostConfig {
+	if vc == nil || info == nil {
 		return nil
 	}
 
+	// Create a copy of the created container's hostconfig.  This is passed in during
+	// container create
+	hostConfig := *vc.HostConfig
```

Comment:
It really can't be nil, but I went ahead and put a check in.  If a container is in the cache, it had to have gone through ContainerCreate.